File: ols-frontend/src/types/review.ts
// 과목의 리뷰 데이터
export interface Review {
    id: number;
    courseName: string;
    author: string; // 리뷰 작성자 이름 (Student 엔티티의 username)
    title: string;
    content: string;
    rating: number; // 별점 (1~5)
    createdAt: string; // ISO 8601 형식의 날짜 문자열
}

// 새로운 리뷰 작성
export interface NewReviewFormData {
    title: string;
    content: string;
    rating: number;
}

export interface ReviewModalProps {
    courseId: number;
    onClose: (reviewSubmitted: boolean) => void;
}

export interface PageInfo {
    page: number;
    totalPages: number;
    totalElements: number;
}

File: ols-backend/src/main/java/com/ols/entity/Review.java
package com.ols.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Table(name = "reviews", uniqueConstraints = {
        @UniqueConstraint(columnNames = {
                "course_id",
                "student_id"
        })
})
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Entity
@Getter
@Setter
public class Review {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String author;

    @Column(nullable = false)
    private String title;

    @Column(nullable = false)
    private String content;

    @Column(nullable = false)
    private Double rating;

    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "course_id", nullable = false)
    private Course course;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "student_id", nullable = false)
    private Student student;

    @PrePersist
    protected void onCreate() {
        this.createdAt = LocalDateTime.now();
    }

    @Builder
    public Review(String author,
                  String title,
                  String content,
                  Double rating,
                  Course course,
                  Student student) {
        this.author = author;
        this.title = title;
        this.content = content;
        this.rating = rating;
        this.course = course;
        this.student = student;
    }

}
